src/figuresClasses.ts

```diff
@@ -1,19 +1,89 @@
-export interface Figure {
+enum Color {
+  Blue = 'blue',
+  Red = 'red',
+  Green = 'green',
+}
+
+enum Shape {
+  Rectangle = 'rectangle',
+  Circle = 'circle',
+  Triangle = 'triangle',
+}
 
+export interface Figure {
+  shape: Shape,
+  color: Color,
+  getArea(): number,
 }
 
-export class Triangle {
+function volueRounding(volue: number): number {
+  return Math.floor(volue * 100) / 100;
+}
 
+function correctSides(...sides: number[]): void {
+  if (sides.some((side) => side <= 0)) {
+    throw new Error('side of the figure must be bigger than zero');
+  }
 }
 
-export class Circle {
+export class Triangle implements Figure {
+  shape = Shape.Triangle;
 
+  constructor(
+    public color: Color,
+    public a: number,
+    public b: number,
+    public c: number,
+  ) {
+    if (a >= b + c || b >= a + c || c >= a + b) {
+      throw new Error('incorrect values for the triangle');
+    }
+    correctSides(a, b, c);
+  }
+
+  getArea(): number {
+    const halfOfPerimeter = (this.a + this.b + this.c) / 2;
+    const squareOfTriangle = Math.sqrt(
+      halfOfPerimeter * (halfOfPerimeter - this.a)
+      * (halfOfPerimeter - this.b)
+      * (halfOfPerimeter - this.c),
+    );
+
+    return volueRounding(squareOfTriangle);
+  }
 }
 
-export class Rectangle {
+export class Circle implements Figure {
+  shape = Shape.Circle;
```

Comment:
do I need to be able to modify it in the instance? Make it read-only

```
const circle = new Circle();

circle.shape = Shape.Rectangle
```

Reply:
fixed